### ISSUE TITLE
invalid RuntimeContext in ServiceFabric silo on range changenotification

### ISSUE THREAD

**Issue by rikbosch** [dotnet/orleans#3057]
Hi,

 We are experiencing some trouble connecting to our cluster running in service fabric. Looking in the silo logs, I see the following exception:

```
[2017-05-26 12:59:16.482 GMT     6    ERROR      103006    EquallyDividedRangeRingProvider    10.0.0.4:20032]    !!!!!!!!!! Local IGrainRingRangeListener Orleans.Streams.ConsistentRingQueueBalancer has thrown an exception when was notified about RangeChangeNotification about old <MultiRange: Size=x68811895, %Ring=40.822%> new <MultiRange: Size=x47EE99B4, %Ring=28.098%>.    
Exc level 0: System.InvalidOperationException: Trying to send a message Request S10.0.0.4:20032:233499509-> #15:  on a silo not from within grain and not from within system target (RuntimeContext is not set to SchedulingContext) RuntimeContext.Current=null TaskScheduler.Current=System.Threading.Tasks.ThreadPoolTaskScheduler
   at Orleans.Runtime.InsideRuntimeClient.SendRequestMessage(GrainReference target, Message message, TaskCompletionSource`1 context, Action`2 callback, String debugContext, InvokeMethodOptions options, String genericArguments)
   at Orleans.Runtime.GrainReference.InvokeMethod_Impl(InvokeMethodRequest request, String debugContext, InvokeMethodOptions options)
   at Orleans.Runtime.GrainReference.InvokeMethodAsync[T](Int32 methodId, Object[] arguments, InvokeMethodOptions options, SiloAddress silo)
   at Orleans.Streams.OrleansCodeGenStreamQueueBalanceListenerReference.QueueDistributionChangeNotification()
   at Orleans.Streams.ConsistentRingQueueBalancer.RangeChangeNotification(IRingRange old, IRingRange now)
   at Orleans.Runtime.ConsistentRing.EquallyDividedRangeRingProvider.RangeChangeNotification(IRingRange old, IRingRange now, Boolean increased)
```

orleans v1.5.0-beta

It looks like a TPL issue, where the notification from ```FabricQueryManager.RegisterPartitionChangeHandler``` comes back on a different context, but I'm not sure..

**Comment by ReubenBond** [comment]
@rikbosch thanks for reporting. I believe you are correct that in your assessment. I'm looking into a fix now.

**Comment by sergeybykov** [comment]
Resolved via #3059.
